pkg/chain/upkeep_key.go

```diff
@@ -0,0 +1,32 @@
+package chain
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+	"strings"
+)
+
+type UpkeepKey []byte
+
+func (u UpkeepKey) BlockKeyAndUpkeepID() (types.BlockKey, types.UpkeepIdentifier, error) {
+	components := strings.Split(string(u), "|")
+	if len(components) == 2 {
+		return types.BlockKey(components[0]), types.UpkeepIdentifier(components[1]), nil
+	}
+	return types.BlockKey(""), types.UpkeepIdentifier(""), fmt.Errorf("%w: missing data in upkeep key", ErrUpkeepKeyNotParsable)
+}
+
+func (u UpkeepKey) String() string {
+	return string(u)
+}
+
+func (u *UpkeepKey) UnmarshalJSON(b []byte) error {
```

Comment:
What's the reason for having this function? It seems like it just implements a default []byte bevavior of json marshaling

Reply:
It's actually no longer needed, it was a hangover from testing my changes mid way through refactoring, I've removed it 👍